### ISSUE TITLE
Aria-hidden needed for modals in v3.3.5

### ISSUE THREAD

**Issue by amberwebb** [twbs/bootstrap#16736]
Based on W3C rules, non-visible elements should have the `aria-hidden=true` attribute. A hidden bootstrap modal should have this attribute, and then the attribute removed when it's visible. `aria-hidden` is not redundant to `display:none` as stated in this PR. https://github.com/twbs/bootstrap/pull/16158 Screen readers do not interpret `display:none`.

http://www.w3.org/TR/aria-in-html/

**Comment by patrickhlauke** [comment]
> Screen readers do not interpret display:none.

yes, they do. In which screen reader/browser combination is this not the case?

**Comment by dak** [comment]
http://www.w3.org/TR/wai-aria/states_and_properties#aria-hidden

`If an element is only visible after some user action, authors MUST set the aria-hidden attribute to true.`


`Note: Authors are reminded that visibility:hidden and display:none apply to all CSS media types; therefore, use of either will hide the content from assistive technologies that access the DOM through a rendering engine. However, in order to support assistive technologies that access the DOM directly, or other authoring techniques to visibly hide content (for example, opacity or off-screen positioning), authors need to ensure the aria-hidden attribute is always updated accordingly when an element is shown or hidden, unless the intent of using off-screen positioning is to make the content visible only to screen reader users and not others.`

Apparently any screen reader that doesn't access the DOM through a rendering engine would need make use of `aria-hidden`.  I don't have enough experience with them to say how prevalent such readers are, although the spec does seem to mandate using aria-hidden rather than relying on display or visibility.

**Comment by patrickhlauke** [comment]
That advice is now fairly outdated. In the upcoming WAI-ARIA 1.1 (currently in working draft) that bit has been removed - see http://www.w3.org/TR/wai-aria-1.1/#aria-hidden

**Comment by stevefaulkner** [comment]
here is start of thread where change in requirement was discussed https://lists.w3.org/Archives/Public/public-pfwg/2014Dec/0035.html

**Comment by patrickhlauke** [comment]
:+1:
